src/main/java/seedu/recipe/model/recipe/IngredientBuilder.java

```diff
@@ -0,0 +1,68 @@
+package seedu.recipe.model.recipe;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.recipe.commons.util.AppUtil.checkArgument;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.NAME_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.parse;
+
+import java.util.HashMap;
+import java.util.List;
+
+import seedu.recipe.logic.parser.Prefix;
+
+/**
+ * Represents a recipe's ingredient in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ */
+public class IngredientBuilder {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredients should follow this format: \n"
+            + "`-a {AMOUNT} -e {ESTIMATED AMOUNT} -n {NAME} `"
+            + "`-cn {COMMON NAME} [-r {REMARKS}] [-s {SUBSTITUTION}]\n"
+            + "i.e. `-a 1 oz. -n butter -r cubed -s margarine`";
```

Comment:
If fields were meant to be optional/variable for this constraints message we should follow the naming convention in our user guide:
![image](https://user-images.githubusercontent.com/88974324/226900790-eeec5cd0-89c3-4cbc-866a-35f05a93192f.png)